new.go

```diff
@@ -222,6 +222,7 @@ func newBuilder(ctx context.Context, store storage.Store, options BuilderOptions
 	var img *storage.Image
 	var err error
 
+	useRegistries := options.Transport == ""
```

Comment:
As mentioned elsewhere, primarily the “helpfully tell the user what registries where tried, and unhelpfully be silent about the underlying error” logic, if it should exist at all, should be moved from `pullImage` to `resolveImage`. Then we _might_ find out that the “search registry list was used” information (not “use [search?] registries” option) is not necessary at all.

And even if we do decide that the “search registry list was used” information is desirable:
- This does not compute the right thing (looking at `ResolveName`, something like `FromImage == "example.com/busybox"` would cause `useRegistries` to be `true`, but `ResolveName` does not actually use the search registry list in that case)
- This is not the right place to compute that information; `ResolveName` _knows_ whether it used the search list or not, so it should just return that as an additional return value, instead of `newBuilder` (_very_ far from where the decision is made, pretty much at the opposite end of the call stack) independently determining the same thing.